packages/@aws-cdk/aws-ecr/lib/public-repository.ts

```diff
@@ -0,0 +1,428 @@
+import { EOL } from 'os';
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import { Lazy, Resource, Stack, Token } from '@aws-cdk/core';
+import { Construct, IConstruct } from 'constructs';
+import { BaseRepositoryProps, IBaseRepository, OnCloudTrailImagePushedOptions } from './base-repository';
+import { CfnPublicRepository, CfnPublicRepositoryProps } from './ecr.generated';
+
+/**
+ * Represents an ECR Public Repository.
+ */
+export interface IPublicRepository extends IBaseRepository {
+
+  /**
+   * The name of the repository.
+   * @attribute
+   */
+  readonly publicRepositoryName: string;
+
+  /**
+   * The ARN of the repository.
+   * @attribute
+   */
+  readonly publicRepositoryArn: string;
```

Comment:
Why not move these up to `IBaseRepository` and use `repositoryName` and `repositoryArn` for both public and private repositories?

Reply:
I cannot do that because linter requires resource name to be of form `${CfnResource.name}Name`.